processor/integrations/integrations.go

```diff
@@ -57,6 +58,51 @@ type PostParametersT struct {
 	Files       map[string]interface{} `json:"files"`
 }
 
+type DeliveryResponseT struct {
+	Status int64  `json:"status"`
+	Body   string `json:"responseBody"`
+}
+
+// This struct represents the datastructure present in Transformer network layer Error builder
+type TransErrorSpecT struct {
+	Message                  string                 `json:"message"`
+	Status                   int                    `json:"status"`
+	StatTags                 map[string]string      `json:"statTags"`
+	DestinationResponse      map[string]interface{} `json:"destinationResponse"`
+	ApiLimit                 map[string]interface{} `json:"apiLimit"`
+	Metadata                 map[string]interface{} `json:"metadata"`
+	ResponseTransformFailure bool                   `json:"responseTransformFailure"`
+	FailureAt                string                 `json:"failureAt"`
+}
+
+type TransErrorT struct {
+	ErrorDetailed TransErrorSpecT `json:"errorDetailed"`
+}
+
+func CollectIntgErrorStats(input []byte, transformationError bool) {
```

Comment:
Deepsource reported this as a control-coupled function: https://deepsource.io/gh/rudderlabs/rudder-server/run/9bce0483-bfc7-4f18-b43a-458da7d9d9cb/go/RVV-A0005

I agree with it, having:
```
func CollectIntgTransformErrorStats(input []byte) {
```

and a 
```
func CollectIntgDestErrorStats(input []byte) {
```

Would be more readable.

Reply:
Got it, addressing this